### DISCUSSION TITLE
Reduce mPoG Parameter and Token Denom to Address Unit Bias and High Cost to Use Network

### DISCUSSION THREAD

**Discussion by CryptoChem0000** [archway-network/discussions#5] [Ideas]
I have received extensive feedback from both users and developers in regards to the high cost of utilizing the network. Specifically, the issue of unit bias and high minimum price of gas (mPoG) have been highlighted, as it noticeably affects user activity and retention.

This feedback has been both direct and indirect, with various community members and general Cosmos contributors discussing the matter. Users have shown reluctance to perform transactions or use Archway decentralized applications due to a perceived high cost in terms of the protocol's native token (ARCH), even though the fiat value of a transaction is generally low (typically less than $0.20).

One proposed solution is to lower the minimum price of gas (mPoG) while allowing developers the option to implement smart contract premiums for setting a higher floor per transaction. This adjustment would lead to a decrease in gas burned per transaction and a reduction in the cost to spam the network. However, this change would allow developers to utilize a smart contract premium to make up for the change in mPoG (if they so choose) and they would actually capture more value as a result. Furthermore, this adjustment would make running public goods on the network more cost-effective on a per tx basis, as the current protocol subsidizes these fees through a fee grant.

Additionally, a reduction in mPoG could enhance the competitiveness of Archway apps compared to similar products. For instance, users seeking to swap assets into ARCH might opt for a decentralized exchange on another network due to the larger fee discrepancy outweighing the difference in slippage. In scenarios involving multiple hops or contract calls, the fee gap expands further, providing users with reason to find cheaper alternatives on different networks that still yield the same outcome (e.g., swapping an asset to ARCH).

Another proposed solution, which would likely be in conjunction with a reduction in mPoG, would be to reduce the token denom. I've floated the idea of changing the denom which in practice would be similar to lowering the initial max supply from 1 billion to 100 million or even 10 million, which would proportionally reduce the amount of ARCH needed for transactions. For example a transaction that requires 1 ARCH currently, would require 0.1 or 0.01 ARCH if the token denom was changed. This change, however, would likely necessitate a hard fork and significant efforts to implement. Therefore, I aim to seek input from the community and core contributors to the network.

**Comment by ewais001** [comment]
Unless there are significant hiccups (which is admittedly very likely), I'm very interested in a hard-fork to move the decimal, and change the supply to 10M instead of 1B. Even though doing so just for psychological reasons seems silly, I DO believe it's legit, and that the dollar value fees are reasonable, and people are just have an aversion to seeing 3 ARCH instead of 0.03 ARCH, even though a $0.20 fee isn't terrible. 

Most of the support for lowering mPoG (including my own) was based on dollar price fees, and as the price of ARCH declined 65%, is no longer an issue imo, but am very curious to see potential growth projections. At this time I just don't think we have 2x the users lined up waiting to use Astrovault, or other Archway dapps, if say a trade cost $0.10 instead of $0.20 in gas, and think that the difference in ARCH burn and dev rewards would be far more substantial than UX benefits or growth at this time.

**Reply by CryptoChem0000** [reply]
I tend to agree that the unit bias is a larger issue than the cost per tx which is more related to the mPoG parameter. I also agree that by just lowering the mPoG, I'm not sure if that will be enough to move the needle and spur more on-chain activity or change current user sentiment regarding the amount of ARCH required per tx. 

In the long-term I think lowering mPoG will be necessary at some point, especially if we see ARCH appreciate over time. However, in the near-term I think the largest issue the network is facing relates to the perceived high cost of submitting a tx (in terms of ARCH spent) which could be alleviated via a hard fork to move the decimal point as mentioned in my original post and your comment.

Now I think the next largest question would pertain to the risks and technical lift associated with a change to the initial supply and mint denom. Would be keen to hear from more technical folks from both the builder's community and Phi Labs team on the potential risks and technical burden this would create for the dapp developers and other contributors to the network.

**Comment by jlm0** [comment]
Thanks for the post outlining the issues around transaction costs. Reducing barriers to usage is key to growing the Archway network.

- If we lower mPoG, what analysis or benchmarks can we use to set the proper minimum? Accounting for effects on developer rewards.
- On the hard fork required for changing the denom, what are the specific technical risks and challenges? Can Phi Labs elaborate on this? How much time and coordination would be needed across developers, validators, etc.?
- You highlighted the unit bias as the main current issue. But if ARCH price rises long-term, wouldn't dollar transaction costs become the new concern?

**Comment by emperorjm** [comment]
Reducing the minimum gas price could be the most viable option. While this might result in dapps earning less per transaction, it could potentially boost the number of user transactions, offsetting this loss. Essentially, if transaction costs decrease tenfold, we might witness a tenfold surge in the volume of transactions. This could lead to an equivalent number of tokens being utilized and subsequently burnt.

On the flip side, the perception of value, influenced by nominal figures, whether it's the price or the volume of a token is a psychological issue I observe both in the stock market and the crypto markets that I trade. For example, take two hypothetical tokens, A and B, both with a market cap of 100 Million. If Token A consists of 1 Billion tokens priced at $0.1 each, and Token B has just 10 Million tokens priced at $10, in my experience most individuals instinctively believe that Token A, priced at $0.1, offers a more affordable and, potentially, a better investment opportunity and they'd likely be more comfortable transacting with Token A given its nominal value.

This is why I think stock splits temporarily succeed with upward price movements as the psychological appeal of a seemingly lower-priced stock often outweighs the logic behind its actual valuation. It's food for thought in understanding market dynamics.

**Comment by zanicar** [comment]
As stated during the governance call on Wednesday, re-denomination at this point is not only unfeasible, **it is impossible**.

Here is a summary of the key takeaways from this call as per my recollection:

It would require the collection, burn, re-issuance and redistribution of the entire Arch supply which also goes against the general blockchain and distributed public ledger ethos and principles. Re-denomination / re-issuance will invalidate the state of any deployed contracts that involve state tracking of tokens... in addition to this any IBC transferred tokens  cannot be re-denominated until they are returned to the native chain, which cannot be guaranteed, thus the network will have to support both going forward.

Regarding the Minimum Price of Gas (MPoG) the conclusion was that this will be a superficial and temporary solution at best, as the primary purpose of tx fees are for security purposes to ensure tx are of legitimate concern. As indicated in comments above it will also diminish the expected earnings of contracts.

It was suggested that the underlying issue or actual problem may be related to current liquidity and that the perceptions of tx fees are merely a symptom of this. Users may be reluctant to spend fees on regular txs as it is difficult for anyone other than validators and dapp developers to obtain additional ARCH to cover these fees, thus available funds are held in reserve for "critical" transactions such as potential re-delegations, etc. It is this phenomenon that may be responsible for low economic velocity and the concerns around txs fees.

If my recollection is incomplete or inaccurate please feel free to indicate / correct as necessary; The purpose here is transparent / open discussion after all!
